src/control/server/storage/provider.go

```diff
@@ -513,12 +513,20 @@ func (p *Provider) ScanBdevs(req BdevScanRequest) (*BdevScanResponse, error) {
 	return scanBdevs(p.log, req, &p.bdevCache, p.bdev.Scan)
 }
 
+func (p *Provider) GetBdevCache() BdevScanResponse {
+	p.RLock()
+	defer p.RUnlock()
+
+	return p.bdevCache
```

Comment:
For my understanding could you tell me why you need to use lock in this function. 
From what I understand, the content of the pointer will not be protected any more after the function return ?

Reply:
OK, thanks for the explanation :)